src/parser/powdr.lalrpop

```diff
@@ -382,7 +382,12 @@ ConstantIdentifier: String = {
     r"%[a-zA-Z_][a-zA-Z$_0-9@]*" => <>.to_string(),
 }
 
-Number: AbstractNumberType = {
-    r"[0-9][0-9_]*" => i128::from_str(&<>.replace('_', "")).unwrap().into(),
-    r"0x[0-9A-Fa-f][0-9A-Fa-f_]*" => i128::from_str_radix(&<>[2..].replace('_', ""), 16).unwrap().into(),
+FieldElement: FieldElement = {
+    r"[0-9][0-9_]*" => i128::from_str(&<>.replace('_', "")).unwrap().try_into().unwrap(),
+    r"0x[0-9A-Fa-f][0-9A-Fa-f_]*" => i128::from_str_radix(&<>[2..].replace('_', ""), 16).unwrap().try_into().unwrap(),
+}
+
+Integer: AbstractNumberType = {
```

Comment:
Why the distinciton here?

Reply:
Because some things like the degree statement get parsed directly to an Integer